### PR TITLE
provider/core: Surfaced ability to restore snapshots from diff view

### PR DESCRIPTION
**Pull request by nwwebb**
Created a restore service which starts a restore snapshot stage in orca. Surfaced the functionality to restore a snapshot from the diff view. PTAL @lwander @duftler @danielpeach

### CODE REVIEW

File: app/scripts/modules/core/snapshot/diff/snapshotDiff.modal.controller.js [hunk 1]
```diff
@@ -75,6 +78,27 @@ module.exports = angular.module('spinnaker.deck.core.snapshot.diff.modal.control
         .then(loadSuccess, loadError);
     };
 
+    this.restoreSnapshot = () => {
+      var submitMethod = () => {
```
Comment (danielpeach) [inline app/scripts/modules/core/snapshot/diff/snapshotDiff.modal.controller.js]: Try not to mix var and let in the same file. (Deck is very inconsistent overall).

File: app/scripts/modules/core/snapshot/snapshot.restore.service.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      return jobs;
+    }
+
+    function restoreSnapshot(app, account, timestamp) {
+      return accountService.getAccountDetails(account).then(function(accountDetail) {
+        let jobs = buildJobs(app, accountDetail, timestamp);
+        return taskExecutor.executeTask({
+          job: jobs,
+          application: app,
+          description: 'Restore Snapshot ' + timestamp + 'of application: ' + app.name + ' for account: ' + accountDetail.name
```
Comment (danielpeach) [inline app/scripts/modules/core/snapshot/snapshot.restore.service.js]: Seems like a good place for a template string: ``Restore Snapshot ${timestamp} of application ${app.name} for account: ${accountDetail.name}``

File: app/scripts/modules/core/snapshot/snapshot.restore.service.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  .module('spinnaker.deck.core.snapshot.restore.service', [
+    require('../task/taskExecutor.js'),
+    require('../utils/lodash.js'),
+    require('../account/account.service.js'),
+    require('../cloudProvider/cloudProvider.registry.js'),
+  ])
+  .factory('snapshotRestorer', function($q, taskExecutor, cloudProviderRegistry,
+                                      accountService, _) {
+
+    function buildJobs(app, accountDetail, timestamp) {
```
Comment (danielpeach) [inline app/scripts/modules/core/snapshot/snapshot.restore.service.js]: Maybe `buildJob`?

File: app/scripts/modules/core/snapshot/snapshot.restore.service.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+module.exports = angular
+  .module('spinnaker.deck.core.snapshot.restore.service', [
+    require('../task/taskExecutor.js'),
+    require('../utils/lodash.js'),
+    require('../account/account.service.js'),
+    require('../cloudProvider/cloudProvider.registry.js'),
+  ])
+  .factory('snapshotRestorer', function($q, taskExecutor, cloudProviderRegistry,
+                                      accountService, _) {
```
Comment (lwander) [inline app/scripts/modules/core/snapshot/snapshot.restore.service.js]: Indentation is off here

File: app/scripts/modules/core/snapshot/snapshot.restore.service.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.deck.core.snapshot.restore.service', [
```
Comment (danielpeach) [inline app/scripts/modules/core/snapshot/snapshot.restore.service.js]: It feels a bit odd to have a service for a particular action, when most services in deck that make API calls are grouped into read and write services.

**Comment by danielpeach** [comment]
LGTM

**Pull request merged**